packages/context/src/ui5-model.ts

```diff
@@ -362,141 +466,319 @@ export async function negotiateVersionWithFetcher(
     );
     isFallback = true;
-  } else if (resolvedVersions[framework]?.[adjustedVersion]) {
-    // version already resolved?
-    const versionDefined = adjustedVersion;
-    adjustedVersion = resolvedVersions[framework]?.[adjustedVersion];
-    if (versionDefined !== adjustedVersion) {
-      isIncorrectVersion = true;
+  } else if (version === UI5_VERSION_S4_PLACEHOLDER) {
+    // handling for S4 placeholder
+    logger.warn(
+      `The version specified as minUI5Version in your manifest.json is not supported by Language Assistant, the latest available version is used instead`
+    );
+    useLatestVersion = true;
+  } else if (version) {
+    const { result, versionDetails } = isVersionSupported(version);
+    if (!result) {
+      if (versionDetails) {
+        // version is out of support in LA, using default version
+        logger.warn(
+          `The version specified as minUI5Version in your manifest.json is not supported by Language Assistant, the fallback version ${fallbackVersion} is used instead`
+        );
+        adjustedVersion = fallbackVersion;
+        isIncorrectVersion = true;
+      } else {
+        // version is incorrect, using latest version
+        logger.warn(
+          `The version specified as minUI5Version in your manifest.json is incorrect, the latest version ${
+            versions["latest"]?.version ?? ""
+          } is used instead`
+        );
+        useLatestVersion = true;
+        isIncorrectVersion = true;
+      }
     }
-  } else if (
-    useLatestVersion ||
-    !(await getVersionInfo(
+  }
+
+  return {
+    adjustedVersion,
+    isFallback,
+    isIncorrectVersion,
+    useLatestVersion,
+  };
+}
+
+/**
+ * Resolves (coerces) requested version based on version map data and CDN data availability.
+ *
+ * @param requestedVersion Version to resolve
+ * @param useLatestVersion Flag whether the latest version should be used
+ * @param framework Framework type
+ * @param versions Version map for the current framework
+ * @param versionInfoJsonFetcher Fetcher function to load remote version data
+ * @param versionLibsFetcher Fetcher function which loads version libs
+ * @param modelCachePath Path to model data cache on disk
+ * @returns Object with resolved version and flag whether the requested version is incorrect and whether the resolved version should be stored in resolution cache
+ * We don't need to store resolution result in cache if test data available
+ */
+async function resolveRequestedVersion(
+  requestedVersion: string,
+  useLatestVersion: boolean,
+  framework: UI5Framework,
+  versions: VersionMapJsonType,
+  versionInfoJsonFetcher: Fetcher<Json>,
+  versionLibsFetcher: Fetcher<Json>,
+  modelCachePath: string | undefined
+): Promise<{
+  resolvedVersion: string;
+  isIncorrectVersion: boolean;
+  remember: boolean;
+}> {
+  let resolvedVersion = requestedVersion;
+  let isIncorrectVersion = false;
+  if (
+    !useLatestVersion &&
+    (await isVersionResourcesAvailable(
```

Comment:
will that restricted check not cause this function to return false due to the fact that there is empty object saved in cache?

Reply:
fixed by above mentioned cache content check condition change